core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -221,11 +298,49 @@ class AdminClient(val time: Time,
   }
 
   def close() {
+    running = false
     client.close()
   }
 
 }
 
+/*
+ * CompositeFuture assumes that the future object in the futures list does not raise error
+ */
+class CompositeFuture[T](time: Time,
+                         defaultResults: Map[TopicPartition, T],
+                         futures: List[RequestFuture[Map[TopicPartition, T]]]) extends Future[Map[TopicPartition, T]] {
+
+  override def isCancelled = false
+
+  override def cancel(interrupt: Boolean) = false
+
+  override def get(): Map[TopicPartition, T] = {
+    get(Long.MaxValue, TimeUnit.MILLISECONDS)
+  }
+
+  override def get(timeout: Long, unit: TimeUnit): Map[TopicPartition, T] = {
+    val start: Long = time.milliseconds()
+    val timeoutMs = unit.toMillis(timeout)
+    var remaining: Long = timeoutMs
+
+    val observedResults = futures.flatMap{ future =>
+      val elapsed = time.milliseconds() - start
+      remaining = if (timeoutMs - elapsed > 0) timeoutMs - elapsed else 0L
+
+      if (future.awaitDone(remaining, TimeUnit.MILLISECONDS)) future.value()
+      else Map.empty[TopicPartition, T]
+    }.toMap
+
+    defaultResults ++ observedResults
+  }
+
+  override def isDone: Boolean = {
+    futures.forall(_.isDone)
+  }
+}
```

Comment:
It's kind of weird to use a ConsumerNetworkClient to implement purgeDataBefore() since this api has nothing to do with consumer.